ksqldb-engine/src/test/java/io/confluent/ksql/schema/ksql/inference/DefaultSchemaInjectorTest.java

```diff
@@ -520,7 +520,15 @@ public void shouldGetValueSchemaWithSchemaId() {
     final ConfiguredStatement<CreateStream> result = injector.inject(csStatement);
 
     // Then:
-    assertThat(result.getStatementText(), containsString("VALUE_SCHEMA_ID=42"));
+    assertThat(result.getStatementText(), is(
+        "CREATE STREAM `cs` ("
+        + "`intField` INTEGER, `bigIntField` BIGINT, "
+        + "`doubleField` DOUBLE, `stringField` STRING, `booleanField` BOOLEAN, "
+        + "`arrayField` ARRAY<INTEGER>, `mapField` MAP<STRING, BIGINT>, "
+        + "`structField` STRUCT<`s0` BIGINT>, `decimalField` DECIMAL(4, 2)) WITH "
+        + "(KAFKA_TOPIC='some-topic', KEY_FORMAT='kafka', VALUE_AVRO_SCHEMA_FULL_NAME='myrecord', "
+        + "VALUE_FORMAT='json_sr', VALUE_SCHEMA_ID=42);"));
```

Comment:
In alignment to my question above. We use `json_sr` format and add `VALUE_AVRO_SCHEMA_FULL_NAME`. Seems to contradict the comment in `DefaultSchemaInjector`?

OrL if we support `VALUE_AVRO_SCHEMA_FULL_NAME` only for AVRO, it should not be added for this test case?

Reply:
Because the returned raw schema was mocked to be `avro` and we check raw schema type to populate the name, that's where's discrepancy is from. We do have check in schema supplier that the format from property must be consistent with the raw format fetched.

Reply:
> Because the returned raw schema was mocked to be `avro`

For this case, shouldn't the expected statement not use `VALUE_FORMAT='avro'`? Where does the `json_sr` come from?

Reply:
`givenValueButNotKeyInferenceSupported` is hardcoded to populate `json_sr` format for value